File: easy/3270_find_the_key_of_the_numbers/solution.py
class Solution:
    def generateKey(self, num1: int, num2: int, num3: int) -> int:
        key = 0

        for div in (1000, 100, 10, 1):
            key = key * 10 + min(num1 // div, num2 // div, num3 // div)

            num1 %= div
            num2 %= div
            num3 %= div

        return key

File: easy/3270_find_the_key_of_the_numbers/solution.cpp
#include <algorithm>

class Solution {
public:
  int generateKey(int num1, int num2, int num3) {
    int key = 0;

    for (int div = 1000; div > 0; div /= 10) {
      key = key * 10 + std::min(num1 / div, std::min(num2 / div, num3 / div));

      num1 %= div;
      num2 %= div;
      num3 %= div;
    }

    return key;
  }
};
